airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/class_types_registry.py

```diff
@@ -56,6 +57,7 @@
     "LimitPaginator": LimitPaginator,
     "ListStreamSlicer": ListStreamSlicer,
     "MinMaxDatetime": MinMaxDatetime,
+    "NoAuth": NoAuth,
```

Comment:
can't we make this part of the default impl registry? it seems odd to explicitly declare no auth in a YAML

Reply:
It sorta already is in a slightly less obvious way: https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/http_requester.py#L58

Reply:
so instead of declaring it here, shouldn't we instead not declare an authenticator wherever we would have used `NoAuth`?

Reply:
We could, I guess it depends on how explicit we want the language to be. I don't think its unheard of that a developer might want to document that the connector doesn't use authentication in the YAML, instead of a expecting future developer to have to understand an underlying low code implementation

Reply:
but i'm also not married to the idea. I can remove it, the rest of the changes are necessary though since options gets passed to every declarative component

Reply:
fair enough. I think for brevity I would err on the side of specifying the presence of something rather than its absence but there's no harm in this change. Out of curiosity what prompted this change?

Reply:
michel was fiddling around with it and was curious why it was a conscious omission. I think I forgot to add it when I was adding no auth during the schema validation change